### PR TITLE
Remove Policy Attachment details from site docs, point to GEP

### PR DESCRIPTION
**Pull request by youngnick**
**What type of PR is this?**
/kind cleanup
/kind documentation

**What this PR does / why we need it**:
Makes Policy Attachment more clearly an experimental pattern and ensures we're oinly documenting that pattern in one place, the GEP.

Split out from #2128 so that we can get this change in sooner.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by shaneutt — APPROVED** [review]
/approve

File: site-src/references/policy-attachment.md [hunk 1]
```diff
@@ -5,744 +5,13 @@ in a standard way as a _Metaresource_. ReferenceGrant
 is an example of this general type of metaresource, but it is far from the only
 one.
 
-This document also defines a concept called _Policy Attachment_, which augments
+Gateway API also document also defines a pattern called _Policy Attachment_, which augments
```
Comment (frankbu) [inline site-src/references/policy-attachment.md:8]:
```suggestion
Gateway API also defines a pattern called _Policy Attachment_, which augments
```

File: site-src/references/policy-attachment.md [hunk 2]
```diff
@@ -5,744 +5,13 @@ in a standard way as a _Metaresource_. ReferenceGrant
...
-
-#### Defaults interacting with other defaults for RetryOnPolicy, value in RetryFilter
-||No default|Namespace default A|Gateway default A|HTTPRoute default A|
-|----|-----|-----|----|----|
-|No default|Value in RetryFilter|Value in RetryFilter|Value in RetryFilter|Value in RetryFilter|
-|Namespace default B|Value in RetryFilter|Value in RetryFilter|Value in RetryFilter|Value in RetryFilter|
-|Gateway default B|Value in RetryFilter|Value in RetryFilter|Value in RetryFilter|Value in RetryFilter|
-|HTTPRoute default B|Value in RetryFilter|Value in RetryFilter|Value in RetryFilter|Value in RetryFilter|
+This pattern is EXPERIMENTAL, and is described in [GEP-713](https://gateway-api.sigs.k8s.io/geps/gep-713/). Please see that document
+technical details.
```
Comment (frankbu) [inline site-src/references/policy-attachment.md:17]:
```suggestion
for technical details.
```

**Comment by youngnick** [comment]
This is a small docs update, so should be ready for an LGTM whenever someone's ready.

**Comment by robscott** [comment]
Thanks @youngnick!

/lgtm

**Pull request merged**